src/main/java/roomescape/user/UserReservationRestController.java

```diff
@@ -0,0 +1,54 @@
+package roomescape.user;
+
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import roomescape.user.domain.Reservation;
+import roomescape.user.repository.ReservationRepository;
+
+@RequiredArgsConstructor
+@RestController
+public class UserReservationRestController {
+
+    private final ReservationRepository reservationRepository;
+
+    @GetMapping("/reservations")
+    public ResponseEntity<List<Reservation>> retrieveReservations() {
+        final List<Reservation> reservations = reservationRepository.findAll();
+
+        return ResponseEntity.ok(reservations);
+    }
+
+    @PostMapping("/reservations")
+    public ResponseEntity<Reservation> persistReservation(@RequestBody final Reservation reservation) {
+        final Long id = reservationRepository.save(reservation);
+        final Optional<Reservation> found = reservationRepository.findById(id);
+
+        return found.map(ResponseEntity::ok)
+                .orElseGet(() -> ResponseEntity.notFound().build());
```

Comment:
```suggestion
        return reservationRepository.findById(id)
                .map(ResponseEntity::ok)
                .orElseGet(() -> ResponseEntity.notFound().build());
```
optional 을 사용하면 가독성을 챙길 수 있다는 장점이 있습니다.
이를 잘 활용해서 리팩터링 해보아요.

Reply:
inline return 하도록 수정했습니다! 😀